sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java

```diff
@@ -17,14 +17,19 @@
 public class InteractiveBrowserCredentialBuilder extends AadCredentialBuilderBase<InteractiveBrowserCredentialBuilder> {
     private Integer port;
     private boolean automaticAuthentication = true;
+    private String redirectURL;
 
     /**
      * Sets the port for the local HTTP server, for which {@code http://localhost:{port}} must be
      * registered as a valid reply URL on the application.
      *
+     * @deprecated Configure the redirectURL as {@code http://localhost:{port}} via
+     * {@link InteractiveBrowserCredentialBuilder#redirectUrl(String)} instead.
+     *
      * @param port the port on which the credential will listen for the browser authentication result
      * @return the InteractiveBrowserCredentialBuilder itself
      */
+    @Deprecated
```

Comment:
Why do we want to deprecate this? Redirect URL will force the user to include the host and the port when they just want to change the port.

Reply:
because, port limits the scope for redirectURL to be only localhost:{port}.
The redirectURL needs to exactly match the URL user specified when registering their custom public client applications.
So, currently it doesn't work for scenarios where registered redirect URL is not localhost.